src/matchers/numbers.lua

```diff
@@ -1,22 +1,128 @@
+local prettyValue = require("src.utils.prettyValue")
+local assertType = require("src.asserts.type")
+
+local INFINITY = math.huge
+
 ---@type lest.Matcher
-local function toBeCloseTo(ctx, received, expected, numDigits) end
+local function toBeCloseTo(ctx, received, expected, numDigits)
+	assertType(received, "number")
```

Comment:
Please do not throw _any_ errors based on the received value. Errors should be thrown only by invalid usage of the matchers in testing instructions, not from the code we're testing.

Also, this does not account for metatables.